src/components/common/LoginModal/LoginModal.tsx

```diff
@@ -0,0 +1,268 @@
+    password: string;
+  };
+  rememberMe: boolean;
+  showPassword: boolean;
+  isLoading: boolean;
+  loggingNetworkError: boolean;
+  loginErrors: {
+    email: string | null;
+    password: string | null;
+  };
+};
+
+type Props = RouteComponentProps;
+
+export class LoginModal extends React.Component<any, State> {
+  static contextType = AppContext;
+  context!: AppContextState;
+  modalTarget = null;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      login: {
+        email: "",
+        password: "",
+      },
+      rememberMe: false,
+      showPassword: false,
+      isLoading: false,
+      loggingNetworkError: false,
+      loginErrors: {
+        email: null,
+        password: null,
+      },
+    };
+  }
+
+  componentDidMount() {
+    this.modalTarget = document.querySelector("#loginContentId");
+    this.disableWindowsScroll();
+  }
+
+  componentWillUnmount() {
+    this.enableWindowScroll();
+  }
+
+  onLoginModalBackgroundClicked = () => {
+    this.closeLoginModal();
+  };
+
+  closeLoginModal = () => {
+    this.context.openLoginModal(false);
+  };
+
+  enableWindowScroll = () => {
+    enableBodyScroll(this.modalTarget);
+  };
+
+  disableWindowsScroll = () => {
+    disableBodyScroll(this.modalTarget);
+  };
+
+  render() {
+    return (
+      <div
+        id={"backgroundModalId"}
+        className={cn(styles.modalBackground)}
+        onClick={(event) => {
+          // @ts-ignore
+          if (event.target.id === "backgroundModalId") {
+            this.onLoginModalBackgroundClicked();
+          }
+        }}
+      >
+        <div id={"loginContentId"} className={styles.modalBox}>
+          <div className={styles.closeButton} onClick={this.closeLoginModal} />
+          <div className={styles.modalContent}>
+            <div className={styles.title}>Sign In</div>
+
+            <div className={styles.signInWithButton}>
+              <div className={cn("fab", "fa-facebook-f", styles.icon)} />
+              Sign in with Facebook
+            </div>
+
+            <div className={styles.signInWithButton}>
+              <div className={cn("fab", "fa-google", styles.icon)} />
+              Sign in with Google
+            </div>
+
+            <div className={styles.signInWithButton}>
+              <div className={cn("fab", "fa-apple ", styles.icon)} />
+              Sign in with Apple
+            </div>
+
+            <div className={styles.orSeparator}>
+              <div className="horizontal-divider" />
+              <div>or</div>
+              <div className="horizontal-divider" />
+            </div>
+
+            {this.state.loggingNetworkError && (
+              <div className={styles.loginAlert}>
+                <i
+                  className={cn(
+                    "fas",
+                    "fa-exclamation-triangle",
+                    styles.errorIcon
+                  )}
+                />
+                We can't find an account with this email address.
```

Comment:
I think we should extract the actual network error and show here. We should have access to it in the catch of the signInClick handler.